### PR TITLE
Add configurable http connection

### LINKED ISSUE #60: Missing client authentication
Trying to connect to https://developer.easee.cloud/docs/signalr I need to supply an auth token. It's unclear how the client options need be configured to set auth headers or supply token.

### PR DESCRIPTION
**Pull request by andig**
Fix https://github.com/philippseith/signalr/issues/60

By passing e.g. an oauth2 client, the http connection connection can securely connect. 

Example:

	ctx := context.Background()
	conn, err := signalr.NewHTTPConnection(ctx, "https://api.easee.cloud/hubs/chargers", signalr.HTTPClientOption(oauth2Client))
	if err != nil {
		return c, err
	}

	client, err := signalr.NewClient(ctx, conn)
	if err != nil {
		return c, err
	}

I would also suggest to remove the `NewHTTPClient` as it has only limited flexibility (breaking change).

### CODE REVIEW

**Comment by philippseith** [comment]
Nice job! Could you please add a test for NewHTTPConnection?

**Comment by philippseith** [comment]
> remove the NewHTTPClient:

As it is only used in tests, I don't bother :-)

**Comment by andig** [comment]
> > remove the NewHTTPClient:
> 
> As it is only used in tests, I don't bother :-)

Maybe I'm doing this wrong- how would you connect to the API above instead?

**Comment by philippseith** [comment]
No, no, you are doing perfectly right. I meant removing the existing NewHTTPClient doesn't bother me.
When writing this package, my main objective was the server. But while writing it, I realised that there should be a simple go client to test the server. And it turned out that server client share a lot of code and are nearly interchangeable.

File: httpconnection.go [hunk 1]
```diff
@@ -88,23 +94,41 @@ func NewHTTPConnection(ctx context.Context, address string, options ...func(*htt
 		// TODO
 	} else if formats = nr.getTransferFormats("WebSockets"); formats != nil {
 		wsURL := reqURL
-		wsURL.Scheme = "ws"
-		ws, _, err := websocket.Dial(ctx, wsURL.String(), nil)
+
+		// switch to wss/443 for secure connection
+		if reqURL.Scheme == "https" {
```
Comment (philippseith) [inline httpconnection.go:101]: nice

**Comment by andig** [comment]
@philippseith do you want to start CI?

**Comment by andig** [comment]
Tests are updated, too and http client removed

File: httpconnection.go [hunk 2]
```diff
@@ -41,27 +42,29 @@ func NewHTTPConnection(ctx context.Context, address string, options ...func(*htt
 	httpConn := &httpConnection{}
 
 	for _, option := range options {
-		option(httpConn)
+		if option != nil {
+			option(httpConn)
```
Comment (philippseith) [inline httpconnection.go:46]: can you please add an error check

**Comment by philippseith** [comment]
Build and Test failed because of a coverage percentage drop. Can you add tests for the added code to fix this?

File: httpconnection.go (hunk 2 above)
Comment (andig) [reply httpconnection.go:46]: arrrrgh- missed this one. coming in new pr

**Pull request merged**